build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/RestTestBasePlugin.java

```diff
@@ -134,20 +135,23 @@ public void apply(Project project) {
             task.systemProperty("tests.system_call_filter", "false");
 
             // Register plugins and modules as task inputs and pass paths as system properties to tests
-            nonInputSystemProperties.systemProperty(TESTS_CLUSTER_MODULES_PATH_SYSPROP, modulesConfiguration::getAsPath);
-            registerConfigurationInputs(task, modulesConfiguration);
-            nonInputSystemProperties.systemProperty(TESTS_CLUSTER_PLUGINS_PATH_SYSPROP, pluginsConfiguration::getAsPath);
-            registerConfigurationInputs(task, extractedPluginsConfiguration);
+            ConfigurableFileCollection modulePath = project.getObjects().fileCollection().from(modulesConfiguration);
```

Comment:
Everything we pass into a CommandLineArgumentProvider needs to be compatible with the configuration cache serialisation. Therefore we wrap configurations in a ConfigurableFileCollection.